### PR TITLE
Upgrade SDKs and Enhance UI with Header Text for Country Code Picker

### PR DESCRIPTION
**Pull request by MrRoy121**
**Summary:**
This pull request introduces significant upgrades and enhancements to the country_code_picker package. Below are the key changes made:

**Updates:**

Flutter SDK: Updated to version 3.22.1 for improved performance and compatibility.
Dart SDK: Updated to version 3.4.1 to leverage the latest language features and improvements.

**Enhancements:**

- **Header Text:** Added a customizable header text to the dialog that appears when the country code picker is clicked, enhancing the user interface and providing better context.

- **Screenshot:** Updated the screenshot image to reflect the new UI changes and provide a clear visual representation of the enhancements.

These updates ensure that the country_code_picker package remains current with the latest SDK versions and provides a better user experience. Please review the changes and consider merging them to enhance the functionality and appearance of the package.

Thank you for your time and consideration!

### CODE REVIEW

File: lib/src/selection_dialog.dart [hunk 1]
```diff
@@ -38,10 +43,16 @@ class SelectionDialog extends StatefulWidget {
     this.favoriteElements, {
     Key? key,
     this.showCountryOnly,
-    this.emptySearchBuilder,
+    this.hideHeaderText = false,
```
Comment (chandrabezzo) [inline lib/src/selection_dialog.dart:46]:
```suggestion
    this.hideHeaderText = true,
```

File: lib/src/selection_dialog.dart [hunk 2]
```diff
@@ -10,15 +10,20 @@ class SelectionDialog extends StatefulWidget {
   final InputDecoration searchDecoration;
   final TextStyle? searchStyle;
   final TextStyle? textStyle;
+  final TextStyle headerTextStyle;
```
Comment (chandrabezzo) [inline lib/src/selection_dialog.dart:13]:
```suggestion
  final TextStyle? headerTextStyle;
```

File: lib/src/selection_dialog.dart [hunk 3]
```diff
@@ -10,15 +10,20 @@ class SelectionDialog extends StatefulWidget {
...
   final double flagWidth;
+  final String headerText;
   final Decoration? flagDecoration;
   final Size? size;
   final bool hideSearch;
   final bool hideCloseIcon;
+  final bool hideHeaderText;
   final Icon? closeIcon;
+  final EdgeInsets topBarPadding;
+  final MainAxisAlignment headerAlignment;
```
Comment (chandrabezzo) [inline lib/src/selection_dialog.dart:26]:
```suggestion
  final EdgeInsets? topBarPadding;
  final MainAxisAlignment? headerAlignment;
```

File: lib/src/selection_dialog.dart [hunk 4]
```diff
@@ -38,10 +43,16 @@ class SelectionDialog extends StatefulWidget {
...
+    this.hideHeaderText = false,
+    this.emptySearchBuilder,this.headerAlignment =  MainAxisAlignment.spaceBetween,
+    this.headerTextStyle =
+        const TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
     InputDecoration searchDecoration = const InputDecoration(),
     this.searchStyle,
     this.textStyle,
+    this.topBarPadding =
+        const EdgeInsets.symmetric(vertical: 5.0, horizontal: 20),
+    this.headerText = "Select Country",
```
Comment (chandrabezzo) [inline lib/src/selection_dialog.dart:55]:
```suggestion
    this.headerText,
```

File: lib/src/selection_dialog.dart [hunk 5]
```diff
@@ -92,12 +104,27 @@ class _SelectionDialogState extends State<SelectionDialog> {
...
+                child: Row(
+                  mainAxisAlignment:widget.headerAlignment,
+                  children: [
+                    !widget.hideHeaderText
+                        ? Text(
+                            widget.headerText,
+                            overflow: TextOverflow.fade,
+                            style: widget.headerTextStyle,
+                          )
+                        : const SizedBox(),
```
Comment (chandrabezzo) [inline lib/src/selection_dialog.dart:118]:
```suggestion
                        : const SizedBox.shrink(),
```

**Comment by MrRoy121** [comment]
@chandrabezzo Sir? any update on my pull request?

**Pull request closed without merging**